libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts

```diff
@@ -3378,6 +3406,235 @@ describe('Lookup component', function () {
     });
+
+    // for testing non-async search args being passed around correctly
+    describe('search args (non-async)', () => {
+      // to test the passing of the 'context' arg
+      describe('context', () => {
+        beforeEach(() => {
+          // setting it to be multiselect is probably not required to test this, but it is the most common use case for using the optional 'context' arg
+          component.setMultiSelect();
+        });
+        describe('search function filters', () => {
+          let searchFilterFunctionSpy: jasmine.Spy;
+
+          it('should return popover context', fakeAsync(() => {
+            // creating the spy function that is pretending to be a search filter without any real functionality, because none is needed to simulate the initial search using a searchFilter
+            searchFilterFunctionSpy = jasmine.createSpy('searchFilterPopover');
+
+            component.customSearchFilters = [searchFilterFunctionSpy];
+            // peforming a search in the popover view should provide the search filter function with a 'popover' context
+            fixture.detectChanges();
+            tick();
+            performSearch('s', fixture);
+            expect(searchFilterFunctionSpy).toHaveBeenCalledWith(
+              's',
+              jasmine.objectContaining({
+                name: jasmine.any(String),
+              }),
+              jasmine.objectContaining({
+                context: 'popover',
+              })
+            );
+          }));
+
+          describe('show more modal', () => {
+            let modalService: SkyModalService;
+
+            beforeEach(fakeAsync(() => {
+              modalService = TestBed.inject(SkyModalService);
+
+              fixture.detectChanges();
+              tick();
+            }));
+
+            // This is necessary as due to modals being launched outside of the test bed they will not
+            // automatically be disposed between tests.
+            afterEach(fakeAsync(() => {
+              // NOTE: This is important as it ensures that the modal host component is fully disposed of
+              // between tests. This is important as the modal host might need a different set of component
+              // injectors than the previous test.
+              modalService.dispose();
+              fixture.detectChanges();
+            }));
+            it('should return modal context', fakeAsync(() => {
+              fixture.detectChanges();
+              tick();
+              const friends: any[] = component.form.controls.friends.value;
+              // creating the spy function that is pretending to be a search filter with real functionality, because a real function filter is needed to be called so that some results may appear in the popover/dropdown so that the "Show more" button may be present so it can be clicked
+              searchFilterFunctionSpy = jasmine
+                .createSpy('searchFilterModal')
+                .and.callFake(
```

Comment:
Instead of calling `.and.callFake`, did you try just doing `searchFilterFunctionSpy = jasmine.createSpy('searchFilterModal').and.callThrough();`?

Reply:
We cannot use `callThrough()`, because we cannot spy on the actual function. 
This is due to the fact that the function is passed into the SkyLookup component through an array.
This means the function does not exist as a function on the fixture, so its functionality cannot be spied on using `callThrough()`.
Therefore, we must use `callFake()` to force the spy to delegate to the provided function.

Reply:
Makes sense.